jfxtras-controls/src/test/java/jfxtras/scene/control/test/CalendarPickerTest.java

```diff
@@ -46,363 +46,363 @@
+		Assert.assertEquals("[2013-01-01]", TestUtil.quickFormatCalendarsAsDate(calendarPicker.calendars()));
+
+		click("#2013-01-03", KeyCode.SHIFT);
+
+		// the last selected value should be set
+		Assert.assertEquals("2013-01-03", TestUtil.quickFormatCalendarAsDate(calendarPicker.getCalendar()));
+		Assert.assertEquals("[2013-01-01, 2013-01-02, 2013-01-03]", TestUtil.quickFormatCalendarsAsDate(calendarPicker.calendars()));
+
+		// extending the range
+		click("#2013-01-05", KeyCode.SHIFT);
+
+		// the last selected value should be set
+		Assert.assertEquals("2013-01-05", TestUtil.quickFormatCalendarAsDate(calendarPicker.getCalendar()));
+		Assert.assertEquals("[2013-01-01, 2013-01-02, 2013-01-03, 2013-01-04, 2013-01-05]", TestUtil.quickFormatCalendarsAsDate(calendarPicker.calendars()));
+		
+		// switching to a single date
+		click("#2013-01-10");
+
+		// the last selected value should be set
+		Assert.assertEquals("2013-01-10", TestUtil.quickFormatCalendarAsDate(calendarPicker.getCalendar()));
+		Assert.assertEquals("[2013-01-10]", TestUtil.quickFormatCalendarsAsDate(calendarPicker.calendars()));
+
+		// selecting a range downwards
+		click("#2013-01-05", KeyCode.SHIFT);
+
+		// the last selected value should be set 
+		Assert.assertEquals("2013-01-05", TestUtil.quickFormatCalendarAsDate(calendarPicker.getCalendar()));
+		Assert.assertEquals("[2013-01-10, 2013-01-09, 2013-01-08, 2013-01-07, 2013-01-06, 2013-01-05]", TestUtil.quickFormatCalendarsAsDate(calendarPicker.calendars()));
+	}
+
+	/**
+	 * 
+	 */
+	@Test
+	public void singleModeWithTime()
+	{
+		TestUtil.runThenWaitForPaintPulse( () -> {
+			calendarPicker.setShowTime(true);
+		});
+		
+		click("#2013-01-01");
+
+		Assert.assertEquals("2013-01-01T00:00:00.000", TestUtil.quickFormatCalendarAsDateTime(calendarPicker.getCalendar()));		
+	}
+
+	/**
+	 * 
+	 */
+	@Test
+	public void singleModeWithTimeSlide()
 	{
-		TestUtil.runThenWaitForPaintPulse( () -> {
-			calendarPicker.setShowTime(true);
-		});
-		
-		// default value is null
-		Assert.assertNull(calendarPicker.getCalendar());
-
-		// set a value
-		final Calendar lCalendar = new GregorianCalendar(2013, 00, 01, 12, 34, 56);
-		TestUtil.runThenWaitForPaintPulse( () -> {
-			calendarPicker.setCalendar( (Calendar)lCalendar.clone() );
-		});
-		Assert.assertEquals("2013-01-01T12:34:56.000", TestUtil.quickFormatCalendarAsDateTime(calendarPicker.getCalendar()));
-
-		// move the hour slider
-		move("#hourSlider > .thumb");
-		press(MouseButton.PRIMARY);
-		moveBy(100,0);		
-		release(MouseButton.PRIMARY);
-		Assert.assertEquals("2013-01-01T20:34:56.000", TestUtil.quickFormatCalendarAsDateTime(calendarPicker.getCalendar()));
-		
-		// move the minute slider
-		move("#minuteSlider > .thumb");
-		press(MouseButton.PRIMARY);
-		moveBy(-50,0);		
-		release(MouseButton.PRIMARY);
-		Assert.assertEquals("2013-01-01T20:23:56.000", TestUtil.quickFormatCalendarAsDateTime(calendarPicker.getCalendar()));

```

Comment:
Don't know why GitHub is telling that I have touched everything..
Actually I've just modified that "23" to "22" because apparently, moving the thumb by 50 pixels is doing 22 minutes.

Same on the other test